pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/realtime/segment/SegmentFlushThresholdComputer.java

```diff
@@ -164,6 +177,17 @@ public int computeThreshold(StreamConfig streamConfig, CommittingSegmentDescript
     return (int) targetSegmentNumRows;
   }
 
+  private long applySegmentSizeVariation(double segmentSizeVariation, long targetSegmentNumRows) {
+    if (segmentSizeVariation > 0.0 && segmentSizeVariation < 1.0) {
+      double variation = (1 - segmentSizeVariation) + 2 * segmentSizeVariation * RANDOM.nextDouble();
+      // Unnecessary to check for negative variation as it qwill always be positive
+      if (variation > 0) {
+        return (long) (targetSegmentNumRows * variation);
+      }
+    }
```

Comment:
Can be simplified since we already bound the variation:
```suggestion
    if (segmentSizeVariation > 0.0) {
      double variation = (1 - segmentSizeVariation) + 2 * segmentSizeVariation * RANDOM.nextDouble();
      return (long) (targetSegmentNumRows * variation);
    }
```